### DISCUSSION TITLE
Is going to a second route possible when waiting for data

### DISCUSSION THREAD

**Discussion by thomasaldershof** [csells/go_router#253] [Q&A] [answered]
Hello all,

I'm trying to do some research on how to navigate and do some deeplinking in the app.
In my example after the user logs in, the home page get some object and if you click on them a detail page opens with that object passed through. Is this possible in go_router (or another package)?

If you take a look at the exmaple [https://gorouter.dev/async-data](https://gorouter.dev/async-data), I would want it to go to the person page without clicking.

Thanks in advance!

**Comment by csells** [comment]
sure. that's pretty much example what the async data example shows.

**Reply by thomasaldershof** [reply]
Unless I miss something in the example you still have to click on a person. So what would it take to make the router automatically go to a person if you provide a family and person ID?

**Comment by csells** [comment]
sure. whenever you have the data, you can do this: `context.go('/family/$fid/person/$pid')`

**Comment by thomasaldershof** [comment]
Would the person page then wait for the family page to finish loading? In my own app I grab the object from a provider on the 'family' page, so that needs to be loaded.

**Comment by csells** [comment, marked as answer]
You can have multiple pages pulling data asynchronously. However, if I were you, I'd use the repository pattern and caching so that only the first fetch actually results in a round-trip over the network.

**Reply by thomasaldershof** [reply]
I'm gonna give it a shot. Thanks!
